src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/BuildPropertyPage.xaml

```diff
@@ -175,37 +175,48 @@
 
   <EnumProperty Name="WarningLevel"
                 DisplayName="Warning level"
-                Description="Specifies the level to display for compiler warnings."
+                Description="Specifies the level to display for compiler warnings. Higher levels produce more warnings, and include all warnings from lower levels."
                 HelpUrl="https://go.microsoft.com/fwlink/?linkid=2146798"
                 Category="ErrorsAndWarnings">
     <EnumValue Name="0"
-               DisplayName="0" />
+               DisplayName="0 - Disable all warnings" />
     <EnumValue Name="1"
-               DisplayName="1" />
+               DisplayName="1 - Severe warnings" />
     <EnumValue Name="2"
-               DisplayName="2" />
+               DisplayName="2 - Less severe warnings" />
     <EnumValue Name="3"
-               DisplayName="3" />
+               DisplayName="3 - Less severe warnings" />
     <EnumValue Name="4"
-               DisplayName="4" />
+               DisplayName="4 - Informational warnings" />
     <EnumValue Name="5"
-               DisplayName="5" />
+               DisplayName="5 - Additional warnings from C# 9" />
```

Comment:
Any reason we shouldn't use the text in #7978 for each of these?

Reply:
Some of those sentences were very long, which I assumed wouldn't work well. But you're right that I should at least try it.

Reply:
Long strings work in the UI better than I expected. I've updated the PR now to:

![image](https://user-images.githubusercontent.com/350947/158721034-d984b565-e1ca-4511-b3a6-da8995769e71.png)

![image](https://user-images.githubusercontent.com/350947/158721076-7de44aac-5aa6-4f84-a117-43457993f02f.png)

![image](https://user-images.githubusercontent.com/350947/158721119-71e166b7-2396-4829-9bb6-c57115f7c8d6.png)

Any concerns?

Reply:
I avoided doing this kind of thing in the property pages for VB PR. You can see those original descriptions here: https://github.com/dotnet/project-system/pull/7928/commits/8b1a5fa2a7d602eb6fedaf6a9d6b36ac995255f5#diff-ae4eb5fe552b47b5719cb7301a0831ca640f3ac659e1a68136bfac7147280f69R444-R458

I originally had descriptions similar to this, but decided against it for a few reasons:
- No clear pattern to follow
  - Here, you used `{value} - {description}` but that is just freehand pattern, not something built into the control
- Without testing, how does the dropdown perform on a small window?
  - Does the item name wrap? Go off the edge of the application? Truncate?
- How easy is it to read/understand from a user perspective?
  - Is it accessible this way?
  - Does it work well with translations/localization?

Instead, I opted to only have the value and relied on the ❓ button to allow the user to read what the values do if they need that information. The linked documentation has better descriptions than we can easily convey in the UI. I also believe we should have a designer weigh in on this concept as these few instances won't be the only dropdowns with long descriptions. We need to have solid guidance to give other teams when implementing their own property pages. Adding @mckennabarlow for input.

Reply:
I like to have a more descriptive value, so I'm in favor of having these texts to let the user know what those numbers translate to. My only concern is, as Yanni mentioned, how it behaves when the window resizes, specially with long strings. I do think it's easier to understand but I'll leave that to be answered by McKenna :).

Reply:
There's a balance here. We want to keep things concise, without removing useless information that could be kept.

The help button is great for learning context around the property's purposes and impact, but is not great as a reference to decode the values we show in the UI.

I also checked and we don't use the `{number} - {description}` pattern elsewhere. But I think it is a natural and obvious pattern. We also need the numbers there to support the description explaining that higher levels include lower levels.

I tested truncation of long values. It currently crops on the RHS, but that's true of all our current controls. We have complete control over this in WPF though, and can wrap long strings if necessary.